### DISCUSSION TITLE
How to deploy a Mediamtx instance to a service?

### DISCUSSION THREAD

**Discussion by andymitch** [bluenviron/mediamtx#3121] [Questions] [answered]
I'm having trouble deploying a Mediamtx instance to flyio, likely due to having to expose multiple ports for a docker container maybe. Does anyone have any experience hosting an instance in the cloud (Fly.io or otherwise)?I'd love to get a thread going with tips and tricks or people's goto solutions.

**Comment by aler9** [comment, marked as answer]
Hello, the [README](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#docker-image) tells how to expose the server, in Docker or in any other containerized service, by using a limited number of ports:

> The --network=host flag is mandatory since Docker can change the source port of UDP packets for routing reasons, and this doesn't allow the RTSP server to identify the senders of the packets. This issue can be avoided by disabling the UDP transport protocol:
>
> docker run --rm -it \
> -e MTX_PROTOCOLS=tcp \
> -e MTX_WEBRTCADDITIONALHOSTS=192.168.x.x \
> -p 8554:8554 \
> -p 1935:1935 \
> -p 8888:8888 \
> -p 8889:8889 \
> -p 8890:8890/udp \
> -p 8189:8189/udp \
> bluenviron/mediamtx
> 
> set MTX_WEBRTCADDITIONALHOSTS to your local IP address.

You can limit the number of ports further by exposing only ports that correspond to the protocol you want to use (i.e. 8554 is for RTSP, 1935 for RTMP, etc... the are all listed in the configuration file).

[a comment by an automated account is not shown]
